### PR TITLE
Add updated ollama models

### PR DESCRIPTION
**Pull request by wacunn**

### CODE REVIEW

**Review by jagapiou — CHANGES_REQUESTED** [review]
Please update this according to the comments in the earlier version of this review (#34).

File: concordia/language_model/ollama_models.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+# Copyright 2024 DeepMind Technologies Limited.
```
Comment (jagapiou) [inline concordia/language_model/ollama_models.py:1]: Nit: please rename this to `ollama_model.py` to be consistent with the other `xyz_model.py` files.

File: concordia/language_model/ollama_models.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+"""Ollama Language Model."""
+
+from concordia.language_model import language_model
+from concordia.utils import measurements as measurements_lib
+from collections.abc import Collection, Sequence
+from langchain import llms
+from langchain.callbacks import manager as callback_manager
+from langchain.callbacks import streaming_stdout
+import re
+from string import Template
```
[the remark this answers is not in the document]
Comment (wacunn) [reply concordia/language_model/ollama_models.py:24]: done

File: concordia/language_model/ollama_models.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+from typing_extensions import override
+
+_PROMPT_TEMPLATE = Template("$system_message\n\n$message")
```
[the remark this answers is not in the document]
Comment (wacunn) [reply concordia/language_model/ollama_models.py:27]: done, and removed import string with this

File: concordia/language_model/ollama_models.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+_TEXT_SYSTEM_MESSAGE = ""
+
+_MULTIPLE_CHOICE_SYSTEM_MESSAGE = """The following is a multiple choice question.
```
[the remark this answers is not in the document]
Comment (wacunn) [reply concordia/language_model/ollama_models.py:31]: done

File: concordia/language_model/ollama_models.py (hunk 1 above)
Comment (wacunn) [reply concordia/language_model/ollama_models.py:1]: I am wondering about this. the reason for the plural is that ollama is a wrapper for many models, and it allows Llama, mistral, gemma, etc. So, ollama is not a singular model per se. Thoughts?

File: concordia/language_model/ollama_models.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+        streaming: bool = False,
+    ):
+        """Initializes the instance.
+
+        Args:
+          model_name (str): The language model to use.
+          system_message (str, optional): The system message to be used in sample methods. Default is an empty string.
+          measurements (Measurements, optional): The measurements object to log usage statistics to. Default is None.
+          channel (str, optional): The channel to write the statistics to. Default is language_model.DEFAULT_STATS_CHANNEL.
+          streaming (bool, optional): If True, enables streaming mode for logging usage statistics. Default is False.
```
[the remark this answers is not in the document]
Comment (wacunn) [reply concordia/language_model/ollama_models.py:62]: removed streaming

File: concordia/language_model/ollama_models.py (hunk 1 above)
Comment (jagapiou) [reply concordia/language_model/ollama_models.py]: It's a module that contains only one class `OllamaLanguageModel`, if it were plural I'd expect to see many in here? To be honest either way I think being consistent with the other module names is preferable.

File: concordia/language_model/ollama_model.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+# limitations under the License.
+
+"""Ollama Language Model."""
+
+import re
+from typing_extensions import override
+
+from concordia.language_model import language_model
+from concordia.utils import measurements as measurements_lib
+from collections.abc import Collection, Sequence
```
Comment (jagapiou) [inline concordia/language_model/ollama_model.py:22]: This is a builtin please move to builtins section.

File: concordia/language_model/ollama_model.py [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ollama Language Model."""
+
+import re
+from typing_extensions import override
```
Comment (jagapiou) [inline concordia/language_model/ollama_model.py:18]: This is third party please move to third party section

File: concordia/language_model/ollama_model.py [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+        self,
+        model_name: str,
+        system_message: str = "",
+        measurements: measurements_lib.Measurements | None = None,
+        channel: str = language_model.DEFAULT_STATS_CHANNEL,
+
+    ):
+        """Initializes the instance.
+        Args:
+            model_name: The language model to use. For more details, see https://github.com/ollama/ollama
```
Comment (jagapiou) [inline concordia/language_model/ollama_model.py:45]: Nit: line too long please add a line break.
https://google.github.io/styleguide/pyguide.html#32-line-length

File: concordia/language_model/ollama_models.py [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    return None
+
+
+class OllamaLanguageModel(language_model.LanguageModel):
+    """Language Model that uses Ollama LLM models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        system_message: str = "",
```
[the remark this answers is not in the document]
Comment (jagapiou) [reply concordia/language_model/ollama_models.py]: SGTM

**Review by jagapiou — COMMENTED** [review]
Looks great Wil, thanks.

Please just fix those three minor things and then rebase as a single commit onto `main` (as you've been doing).

I'll then approve and it will get automatically pulled to our internal source-of-truth, merged, and then pushed back out with you credited.

**Review by jagapiou — APPROVED** [review]

**Pull request merged**